### PR TITLE
ci: Implement a prim and proper release workflow

### PR DESCRIPTION
**Pull request by BurningEnlightenment**
Always build the GH Pages from the default branch to ensure that the documentation is always built with the _latest_ SMV and not e.g. a patch version for an older release line.

Move the actual build pipeline into its own reusable workflow in order to allow fine grained builds of the documentation and release artifacts.

Add a CI job to create a GH release draft for a new tag which automatically attaches the released build artifacts.

@Holzhaus I'm sorry it took so long, but the last weeks have been exceptionally busy for me 😓. I renamed the release workflow file i.e. you'll need to update the trusted publisher settings ([PyPI docs][pypi-docs]). The GitHub environmnent for non-test PyPI should probably be configured to require two approvals and a reasonable mandatory wait timer. After this has been reviewed and merged we should release v0.2.5a0 to thoroughly test the workflow.

I'm available throughout the weekend and the next week, so please proceed at your own convenience. Just make sure to @ me as I'm not eyeing the GitHub notification bell.

[pypi-docs]: https://docs.pypi.org/trusted-publishers/adding-a-publisher/#github-actions

### CODE REVIEW

**Comment by BurningEnlightenment** [comment]
Ah, it seems like I need to push this directly to the upstream repository to allow the new CI to run.

**Review by Holzhaus — APPROVED** [review]
> @Holzhaus I'm sorry it took so long, but the last weeks have been exceptionally busy for me 😓. 

@BurningEnlightenment Don't worry, I can relate to this ;-)

> renamed the release workflow file i.e. you'll need to update the trusted publisher settings ([PyPI docs](https://docs.pypi.org/trusted-publishers/adding-a-publisher/#github-actions)).

Done.

> The GitHub environmnent for non-test PyPI should probably be configured to require two approvals and a reasonable mandatory wait timer. After this has been reviewed and merged we should release v0.2.5a0 to thoroughly test the workflow.

There is no way configure a specific number of reviewers, at least that I know of. But review from the maintainers is required.

File: .github/workflows/branch-build.yaml
Comment (BurningEnlightenment) [reply .github/workflows/branch-build.yaml:5]: I've entertained the thought to add branches like `release/v0.1.x` and later on `release/v0.2.x`, `release/v1.x.x` `release/v1.0.x`. These branches would act as `main` branch for earlier release lines, so one easily knows where to start if a bug needs to be fixed. This is usually only relevant after a release with breaking changes where users might not want to upgrade directly. It's your call though :)

**Comment by BurningEnlightenment** [comment]
> There is no way configure a specific number of reviewers, at least that I know of.

Nevermind, it seems like I conflated this with GitLab.

**Pull request merged**
